File: AgoraChat-ObjectiveC/AgoraChat/Code/Utilities/Category/NSObject+AlertController.m
//
//  NSObject+AlertController.m
//  AgoraChat
//
//  Created by liu001 on 2022/3/23.
//  Copyright © 2022 easemob. All rights reserved.
//

#import "NSObject+AlertController.h"
#import "UIViewController+Util.h"

@implementation NSObject (AlertController)
- (void)_showAlertController:(UIAlertController *)aAlert
{
    UIAlertAction *okAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"common.ok", @"OK") style:UIAlertActionStyleCancel handler:nil];
    [aAlert addAction:okAction];
    
    UIViewController *currentViewController = [aAlert currentViewController];
    aAlert.modalPresentationStyle = 0;
    [currentViewController presentViewController:aAlert animated:YES completion:nil];
}

- (void)showAlertWithMessage:(NSString *)aMsg
{
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"" message:aMsg preferredStyle:UIAlertControllerStyleAlert];
    [self _showAlertController:alertController];
}

- (void)showAlertWithTitle:(NSString *)aTitle
                   message:(NSString *)aMsg
{
    UIAlertController *alertController = [UIAlertController alertControllerWithTitle:aTitle message:aMsg preferredStyle:UIAlertControllerStyleAlert];
    [self _showAlertController:alertController];
}

@end

File: AgoraChat-Swift/AgoraChat-Swift/code/Utilities/Category/NSObject+AlertController.swift
//
//  NSObject+AlertController.swift
//  AgoraChat-Swift
//
//  Created by 冯钊 on 2022/10/12.
//

import UIKit

@objc extension NSObject {
    @objc func showAlert(title: String? = nil, message: String) {
        guard let rootViewController = UIWindow.keyWindow?.rootViewController else {
            return
        }
        let controller = UIAlertController(title: title, message: message, preferredStyle: .alert)
        controller.addAction(UIAlertAction(title: LocalizedString.Ok, style: .cancel))
        controller.modalPresentationStyle = .fullScreen
        rootViewController.present(controller, animated: true)
    }
}
